### DISCUSSION TITLE
Conversation with history in LangServe

### DISCUSSION THREAD

**Discussion by petrishutin** [langchain-ai/langserve#286] [Q&A] [answered]
Hi, everyone!

Could you tell me if there is a way to build application with LangServe that can keep the context of conversation? It is not mentioned in official docs, and I did not find any examples. Is there a way to use RunnableWithMessageHistory in LangServe application?

I will be grateful if some one give a link to an example of code.

Thank you!

**Comment by petrishutin** [comment]
I found the discussion in  [issues](https://github.com/langchain-ai/langserve/issues/275). So I am waiting [eyurtsev](https://github.com/eyurtsev) answer

**Comment by eyurtsev** [comment]
Example is here: 

https://github.com/langchain-ai/langserve/pull/300/files

It's reverted until a PR lands in langchain: 

https://github.com/langchain-ai/langchain/pull/14351

If you're blocked on this, you can copy paste the updates to the RunnableWithMessageHistory into your own code base.

I'll update this issue once relevant code has been merged and released

**Comment by eyurtsev** [comment, marked as answer]
Example here:

Using `session_id` passed via config:

https://github.com/langchain-ai/langserve/tree/main/examples/chat_with_persistence

Using `conversation_id` and a `user_id`. In this example `user_id` is lifted off a cookie, but you can swap with your own way of identifying the user (e.g., jwt)

https://github.com/langchain-ai/langserve/tree/main/examples/chat_with_persistence_and_user

These require langchain_core >= 0.1.0

**Reply by petrishutin** [reply]
Thank you very much!
